ServerCore/Pages/Hints/Edit.cshtml.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Hints
+{
+    public class EditModel : EventSpecificPageModel
+    {
+        private readonly ServerCore.DataModel.PuzzleServerContext _context;
+
+        public EditModel(ServerCore.DataModel.PuzzleServerContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Hint Hint { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            Hint = await _context.Hints.FirstOrDefaultAsync(m => m.Id == id);
+
+            if (Hint == null)
+            {
+                return NotFound();
+            }
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            ModelState.Remove("Hint.Puzzle");
```

Comment:
What is this?

Reply:
Added a comment (this is because ASP.NET doesn't round-trip the Puzzle property and so validation fails because it thinks it's null. This line removes the check for it)